### ISSUE TITLE
extreme redundency

### ISSUE THREAD

**Issue by bdhrs** [digitalpalitools/inflection-generator#2]
there is extreme redundancy in the database i.e. multiple version of the same inflection being generated by a different headword, but having the same stem + pattern combination. 

for example, all these headwords have the same stem + pattern = * ta pron
naṃ, ne 1, ne 2, sa 2, sā, so 1, ta 2, tā 1, tad, tāhi 1, tāhi 2, taṃ 1, taṃ 2, tamhā, tamhi, tāni 1, tāni 2, tāsaṃ 1, tāsaṃ 2, tasmā 1, tasmiṃ 1, tasmiṃ 2, tassa 1, tassa 2, tassa 3, tassā 1, tassā 2, tassaṃ, tāsu, tat, tato 1, tato 2, tāya 1, tāya 3, tāyaṃ, te 1, te 2, tehi 1, tehi 2, tena 1, tesaṃ 1, tesaṃ 2, tesaṃ 3, tesu

this results in **184** exact copies of _tesaṃ_, all generated from the same stem + pattern (*ta pron)

**Comment by bdhrs** [comment]
query tesaṃ - sort by pāli. you can see all the different headwodrs generating the same declension. 
![tesaṃ](https://user-images.githubusercontent.com/64521731/107872108-b8a36880-6ecd-11eb-942d-79c6f3665822.png)

**Comment by parthopdas** [comment]
**TL;DR** - this isn't redundancy. we'll optimize db later depending on the need.

SELECT Count(DISTINCT pāli) FROM 'all_words' => 542262
SELECT Count(pāli) FROM 'all_words'  => 1357392

that is the level dups in pāli. that dup is not the same as redundancy.
we'll optimize the db later on. even the db design is highly suboptimal -> i made it so you can easily read and spot check instead of optimizing and making it unreadable

**Comment by bdhrs** [comment]
let me try and explain it another way. the redundancy is coming from my side, not from the db. 

for example. i have multiple words in pāli1 which all have the same stem and pattern. here are 12 words, all with same stem and pattern (* kaci pron)
![kaci pron](https://user-images.githubusercontent.com/64521731/107879326-2bc4d300-6efe-11eb-9dfe-08a337870355.png)

so that produces exactly the same pattern of declensions 12 times, where once is enough for that stem + pattern combo.  

that's the kind of redundancy i mean.
